diff --git a/common.gypi b/common.gypi
index d1411ed..d87205e 100644
--- a/common.gypi
+++ b/common.gypi
@@ -278,6 +278,9 @@
           ['_type=="static_library" and OS=="solaris"', {
             'standalone_static_library': 1,
           }],
+          ['OS=="openbsd"', {
+            'ldflags': [ '-Wl,-z,wxneeded' ],
+          }],
         ],
         'conditions': [
           [ 'target_arch=="ia32"', {
diff --git a/deps/cares/include/ares.h b/deps/cares/include/ares.h
index cd94d4f..c23ea3e 100644
--- a/deps/cares/include/ares.h
+++ b/deps/cares/include/ares.h
@@ -38,7 +38,7 @@
    require it! */
 #if defined(_AIX) || defined(__NOVELL_LIBC__) || defined(__NetBSD__) || \
     defined(__minix) || defined(__SYMBIAN32__) || defined(__INTEGRITY) || \
-    defined(ANDROID) || defined(__ANDROID__)
+    defined(ANDROID) || defined(__ANDROID__) || defined(__OpenBSD__)
 #include <sys/select.h>
 #endif
 #if (defined(NETWARE) && !defined(__NOVELL_LIBC__))
